app/client/components/consent/PrivacySettings.tsx

```diff
@@ -101,58 +66,87 @@ const buttonStyles = css`
   cursor: pointer;
   position: relative;
   background-color: ${palette.yellow.medium};
-  flex-grow: 1;
+  flex: 1;
   margin: 0 6px;
 `;
 
-const tickIconStyles = css`
-  height: 20px;
-  width: 20px;
-  flex-grow: 0;
-`;
-
 interface State {
-  purposes: PurposeList;
+  iabPurposes: { [key: number]: boolean | null };
+}
+
+interface IabPurpose {
+  id: number;
+  name: string;
+  description: string;
+}
+
+interface IabFeature {
+  id: number;
+  name: string;
+  description: string;
+}
+
+interface IabVendor {
+  id: number;
+  name: string;
+  policyUrl: string;
+  purposeIds: number[];
+  legIntPurposeIds: number[];
+  featureIds: number[];
+}
+
+interface ParsedIabVendor {
+  id: number;
+  name: string;
+  policyUrl: string;
+  purposeIds: number[];
+  legIntPurposeIds: number[];
+  featureIds: number[];
+  description: React.ReactNode;
+}
+
+interface IabVendorList {
+  vendorListVersion: number;
+  lastUpdated: string;
+  purposes: IabPurpose[];
+  features: IabFeature[];
+  vendors: IabVendor[];
+}
+
+interface ParsedIabVendorList extends IabVendorList {
+  vendors: ParsedIabVendor[];
 }
 
 export class PrivacySettings extends Component<{}, State> {
+  private iabVendorList?: ParsedIabVendorList;
+
   constructor(props: {}) {
     super(props);
 
-    this.state = { purposes };
+    this.state = { iabPurposes: {} };
   }
 
   public componentDidMount(): void {
+    fetch(
+      "https://assets.guim.co.uk/data/vendor/4f4a6324c7fe376c17ceb2288a84a076/cmp_vendorlist.json"
```

Comment:
Wrong link. The constant link is:

https://www.theguardian.com/commercial/cmp/vendorlist.json

Reply:
👍

Reply:
I've was trying to use the right link but hit issues with CORS. Let's talk about this tomorrow